python/modules/IcePy/Slice.cpp

```diff
@@ -72,153 +65,76 @@ IcePy_loadSlice(PyObject* /*self*/, PyObject* args)
             return nullptr;
         }
     }
-    catch (const IceInternal::BadOptException& ex)
-    {
-        PyErr_Format(PyExc_RuntimeError, "error in Slice options: %s", ex.what());
-        return nullptr;
-    }
-    catch (const IceInternal::APIException& ex)
+    catch (const std::exception& ex)
     {
         PyErr_Format(PyExc_RuntimeError, "error in Slice options: %s", ex.what());
         return nullptr;
     }
 
     vector<string> cppArgs;
-    Ice::StringSeq includePaths;
     bool debug = false;
-    bool all = false;
-    if (opts.isSet("D"))
+
+    for (const auto& arg : opts.argVec("D"))
```

Comment:
No need for the extra `if (opts.isSet` the `argVec` call should return an empty vector if not set.